catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.async.data.impl;
+
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.DEFAULT_MIME_TYPE;
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.DEFAULT_NAME;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.function.Supplier;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.resource.Resource;
+
+public class LazyProcessResourceImpl implements ProcessResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LazyProcessResourceImpl.class);
+
+    private URI uri;
+
+    private String name;
+
+    private String mimeType;
+
+    private long size = UNKNOWN_SIZE;
+
+    private InputStream inputStream;
+
+    private String qualifier;
+
+    private boolean isModified = false;
+
+    private boolean isResourceLoaded = false;
+
+    private Supplier<Resource> resourceSupplier;
+
+    private String metacardId;
+
+    /**
+     * Creates a {@link ProcessResource} with a {@link Supplier} used to get the a resource.
+     * The resource will not be loaded until the first time one of the following fields is accessed:
+     * inputStream
+     * mimeType
+     * name
+     *
+     * @param metacardId       schema specific part of {@link URI}, throws {@link IllegalArgumentException} if empty or null
+     * @param resourceSupplier a {@link Supplier} used to load the resource
```

Comment:
Might want to link `{@link Resource}` in this javadoc

Reply:
Linked